src/logic/__test__/calculate.test.js

```diff
@@ -0,0 +1,48 @@
+import calculate from '../calculate';
+
```

Comment:
- [ ] You will then need to write separate tests for the `calculate.js` using react testing library to simulate user interactions to test all the scenarios involved:

- If the buttonName is 'AC', the function resets the calculator data object to its initial state by setting total, next, and operation to null.
- If buttonName is a number (checked using the isNumber function), the function updates the calculator data object based on the current state and the button pressed. Depending on whether there is an ongoing operation or not, the next and total properties are updated to build the number to be operated on.
- If buttonName is '.', the function handles decimal input. It ensures that a decimal point is added to either the next or total property only if it's not already present.
- If buttonName is '=', the function performs the calculation based on the current state (if there's a pending operation) and updates the total property while clearing next and operation.
- If buttonName is '+/-', the function toggles the sign of the current number (either in next or total) by multiplying it by -1.
- If buttonName is any other operation (e.g., +, -, *, /), the function handles the logic for applying the operation. It considers cases like when an operation is pressed immediately after = or when there's already an ongoing operation.



Check out an example of how to write the tests:

```
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from './Calculator'; // Assuming you have a Calculator component

describe('Calculator', () => {
  it('resets calculator data when AC button is clicked', () => {
    const { getByText, getByTestId } = render(<Calculator />);
    
    const totalDisplay = getByTestId('total-display');
    const nextDisplay = getByTestId('next-display');
    const operationDisplay = getByTestId('operation-display');
    
    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('AC'));

    expect(totalDisplay).toHaveTextContent('Total: null');
    expect(nextDisplay).toHaveTextContent('Next: null');
    expect(operationDisplay).toHaveTextContent('Operation: null');
  });

  it('handles number input and operations correctly', () => {
    const { getByText, getByTestId } = render(<Calculator />);
    
    const totalDisplay = getByTestId('total-display');
    const nextDisplay = getByTestId('next-display');
    const operationDisplay = getByTestId('operation-display');
    
    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));
    
    expect(totalDisplay).toHaveTextContent('Total: 3');
    expect(nextDisplay).toHaveTextContent('Next: null');
    expect(operationDisplay).toHaveTextContent('Operation: null');
  });

  it('handles decimal input correctly', () => {
    const { getByText, getByTestId } = render(<Calculator />);
    
    const nextDisplay = getByTestId('next-display');
    
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('.'));
    fireEvent.click(getByText('5'));
    
    expect(nextDisplay).toHaveTextContent('Next: 2.5');
  });

  it('handles +/- button correctly', () => {
    const { getByText, getByTestId } = render(<Calculator />);
    
    const nextDisplay = getByTestId('next-display');
    
    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('+/-'));
    
    expect(nextDisplay).toHaveTextContent('Next: -1');
  });

  it('performs calculations correctly', () => {
    const { getByText, getByTestId } = render(<Calculator />);
    
    const totalDisplay = getByTestId('total-display');
    const nextDisplay = getByTestId('next-display');
    const operationDisplay = getByTestId('operation-display');
    
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));
    
    fireEvent.click(getByText('-'));
    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('='));
    
    expect(totalDisplay).toHaveTextContent('Total: 4');
    expect(nextDisplay).toHaveTextContent('Next: null');
    expect(operationDisplay).toHaveTextContent('Operation: null');
  });

  // Add more test cases for different scenarios...

});
```